app/src/org/commcare/activities/FormAndDataSyncer.java

```diff
@@ -60,7 +66,7 @@ public void processAndSendForms(final SyncCapableCommCareActivity activity,
                                     FormRecord[] records,
                                     final boolean syncAfterwards,
                                     final boolean userTriggered) {
-
+        HiddenPreferences.updateLastUploadSyncAttemptTime();
```

Comment:
Just curious - Why isn't this also checking for the syncAfterwards flag? odd that it's passed in if it's always true.

If line 51 only ever results in a call to this method, i'd do the wrapping here, and ignore the edge case above, since it only occurs when there are no unsent forms and in that case a sync attempt would occur anyway if things were user triggered, which would naturally set the flag correctly

Reply:
@ctsims Hey, it's true that the only call to this method is from `checkAndStartUnsentFormsTask` . I have made this method `private` now to reflect the same and moved the `updateLastUploadSyncAttemptTime` call to the calling function `checkAndStartUnsentFormsTask` since we are no longer updating this time now from the restore sync and therefore we need to account for the edge case when there are no unsent forms as well.